### ISSUE TITLE
StackOverflawException with default constructed value (Since 4.142.0)

### ISSUE THREAD

**Issue by ryoasai** [angularsen/UnitsNet#1118]
**Describe the bug**
StackOverflawException when getting value from default constructed value object. See reproduction sample below.

**To Reproduce**
Steps to reproduce the behavior (just an example):
1. Add nuget UnitsNet 4.142.0 or later to a .NET 6 project
2. Build using Visual Studio 2022 Community v17.3.0

```C#
        [Fact]
        public void Init_DefaultConstruct()
        {
            Length x = default;
            Assert.Equal(0.0, x.Micrometers); // StackOverflowExeption when accessing X.Micrometers property.
        }

```
**Expected behavior**
A value object should not cause a StackOverflowExeption for default value and it should give 0 as was previous version.

**Comment by ryoasai** [comment]
Actually, I noticed this problem as I utilize your library to represent point with length unit.
The Point is struct and the X, Y property might be default value and causes the same StackoverflowException.

As C# 10 allow to define default constructs for structs, so I might define default constructor to explicitly initialize Length.Zero, but this workaround is only possible for C#10 and, might not not always safe as value object might be initialized in generics class.
  
  ```C#
public readonly struct Point : IEquatable<Point>, IShape<Point>
    {
        public const double MaxTolerance = 0.00001;
        public static readonly Point Origin = new Point();

        public Length X { get; }
        public Length Y { get; }

        public Point(double x, double y) : this(Length.FromMicrometers(x), Length.FromMicrometers(y))
        {
        }

        [JsonConstructor]
        public Point(Length x, Length y)
        {
            X = x;
            Y = y;
        }

        bool IEquatable<Point>.Equals(Point other) => Equals(other);

        public bool Equals(in Point other) => Equals(other, MaxTolerance, ComparisonType.Absolute);

        public bool Equals(in Point other, double tolerance, ComparisonType comparisonType) =>
            X.Equals(other.X, tolerance, comparisonType) && Y.Equals(other.Y, tolerance, comparisonType);


        public override bool Equals(object? obj) => obj is Point other && Equals(other);

        public static bool operator ==(in Point left, in Point right) => left.Equals(right);

        public static bool operator !=(in Point left, in Point right) => !left.Equals(right);

        public static Point operator +(in Point point) => point;

        public static Point operator -(in Point point) => Negate(point);

        public static Point operator +(in Point left, in Point right) => Add(left, right);
        public static Point operator -(in Point left, in Point right) => Subtract(left, right);


        public static Point operator *(double scale, in Point point) => point.Scale(scale);
        public static Point operator *(in Point point, double scale) => point.Scale(scale);

        public override int GetHashCode() => HashCode.Combine(X, Y);

        public override string ToString() => $"Point({X}, {Y})";

        public Point Move(double dx, double dy) =>
            Move(Length.FromMicrometers(dx), Length.FromMicrometers(dy));

        public Point Move(Length dx, Length dy)
            => new Point(X + dx, Y + dy);

        public Point Translate(in Point newOrigin)
            => Move(-newOrigin.X, -newOrigin.Y);


        public Point MirrorY() => new Point(-X, Y);
        public Point MirrorX() => new Point(X, -Y);

        public Point Rotate(RotateAngle angle) =>
            angle switch
            {
                RotateAngle.Deg0 => this,
                RotateAngle.Deg90 => new Point(-Y, X),
                RotateAngle.Deg180 => new Point(-X, -Y),
                RotateAngle.Deg270 => new Point(Y, -X),
                _ => throw new ArgumentOutOfRangeException(nameof(angle), angle, null)
            };

        public Point Scale(double scale) => new Point(X * scale, Y * scale);

        public Point Apply(in Transform transform) =>
            transform.Apply(this);

        public static Point Negate(in Point point) => new Point(-point.X, -point.Y);

        public static Point Plus(in Point item) => item;

        public static Point Add(Point left, Point right) => new Point(left.X + right.X, left.Y + right.Y);

        public static Point Subtract(Point left, Point right) => new Point(left.X - right.X, left.Y - right.Y);
    }
```

**Comment by lipchev** [comment]
@tmilnthorp I haven't tested this but looking at the code, my feeling is that this is related to [this line here](https://github.com/angularsen/UnitsNet/blob/cc803319ba4af173a0f81146c6c3c39aa72fa439/UnitsNet/GeneratedCode/Quantities/Length.g.cs#L1375)- where I think we should not be using the nullable field but rather the Unit property (returning BaseUnit by default).

**Comment by angularsen** [comment]
[Release UnitsNet/4.144.0 · angularsen/UnitsNet](https://github.com/angularsen/UnitsNet/releases/tag/UnitsNet%2F4.144.0)

**Comment by ryoasai** [comment]
Thank you for your quick fix!

I'll verify the fix once it will be published at nuget repo.

**Comment by angularsen** [comment]
Build hung, requeued.
https://ci.appveyor.com/project/angularsen/unitsnet/history

**Comment by ryoasai** [comment]
I verified the fix with 4.144.0

**Comment by angularsen** [comment]
Great! Thank you for reporting back 👍
